include/ddc/kernels/splines/spline_builder_2d_batched.hpp

```diff
@@ -0,0 +1,287 @@
+                derivs_min2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type2, Layout, memory_space>> const
+                derivs_max2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                mixed_derivs_min1_min2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                mixed_derivs_max1_min2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                mixed_derivs_min1_max2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                mixed_derivs_max1_max2) const
+{
+    const std::size_t nbc_xmin = m_spline_builder1.s_nbc_xmin;
+    const std::size_t nbc_xmax = m_spline_builder1.s_nbc_xmax;
+    const std::size_t nbc_ymin = m_spline_builder2.s_nbc_xmin;
+    const std::size_t nbc_ymax = m_spline_builder2.s_nbc_xmax;
+
+    // TODO: perform computations along dimension 1 on different streams ?
+    // Spline1-transform derivs_min2 (to spline1_deriv_min)
+    ddc::Chunk spline1_deriv_min_alloc(
+            m_spline_builder_deriv1.spline_domain(),
+            ddc::KokkosAllocator<double, MemorySpace>());
+    auto spline1_deriv_min = spline1_deriv_min_alloc.span_view();
+    auto spline1_deriv_min_opt = std::optional(spline1_deriv_min.span_cview());
+    if constexpr (BcXmin1 == ddc::BoundCond::HERMITE) {
+        m_spline_builder_deriv1(
+                spline1_deriv_min,
+                *derivs_min2,
+                mixed_derivs_min1_min2,
+                mixed_derivs_max1_min2);
+    } else {
+        spline1_deriv_min_opt = std::nullopt;
+    }
+
+    // Spline1-transform vals (to spline1)
+    ddc::Chunk spline1_alloc(
+            m_spline_builder1.spline_domain(),
+            ddc::KokkosAllocator<double, MemorySpace>());
+    ddc::ChunkSpan spline1 = spline1_alloc.span_view();
+
+    m_spline_builder1(spline1, vals, derivs_min1, derivs_max1);
+
+    // Spline1-transform derivs_max2 (to spline1_deriv_max)
+    ddc::Chunk spline1_deriv_max_alloc(
+            m_spline_builder_deriv1.spline_domain(),
+            ddc::KokkosAllocator<double, MemorySpace>());
+    auto spline1_deriv_max = spline1_deriv_max_alloc.span_view();
+    auto spline1_deriv_max_opt = std::optional(spline1_deriv_max.span_cview());
+    if constexpr (BcXmax1 == ddc::BoundCond::HERMITE) {
+        m_spline_builder_deriv1(
+                spline1_deriv_max,
+                *derivs_max2,
+                mixed_derivs_min1_max2,
+                mixed_derivs_max1_max2);
+    } else {
+        spline1_deriv_max_opt = std::nullopt;
+    }
+
+    // Spline2-transform spline1
+    m_spline_builder2(
+            spline,
+            spline1.span_cview(),
+            std::optional(spline1_deriv_min_opt),
+            std::optional(spline1_deriv_max_opt));
```

Comment:
It is already optional, should work
```suggestion
            spline1_deriv_min_opt,
            spline1_deriv_max_opt);
```